### DISCUSSION TITLE
API and complex filters

### DISCUSSION THREAD

**Discussion by jsjoshua** [nocodb/nocodb#8633] [Q&A] [answered]
Is there a way to use complex logic in the "where" filter?  I can use "~and" or "~or" and I can even use a single nested "~or), like where=(column1,eq,a)~and((column2,eq,b)~or(column3,eq,c)). But I get an error every time I try to go further, e.g.:  where=(column1,eq,a)~and((column2,eq,b)~or(column3,eq,c))~and((column4,eq,d)~or(column5,eq,e)).  Thanks.

**Comment by mertmit** [comment, marked as answer]
It seems there is a bug while parsing groups after first one, I've created an issue to follow this https://github.com/nocodb/nocodb/issues/8835.

Thanks for reporting, lets use issue instead of this discussion to avoid duplicates.
